File: Java/exercicio11.java
abstract class Funcionario {
    abstract double calcularSalario();
}

class FuncionarioHorista extends Funcionario {
    int horasTrabalhadas;
    double valorPorHora;

    FuncionarioHorista(int horasTrabalhadas, double valorPorHora) {
        this.horasTrabalhadas = horasTrabalhadas;
        this.valorPorHora = valorPorHora;
    }

    double calcularSalario() {
        return horasTrabalhadas * valorPorHora;
    }
}

class FuncionarioAssalariado extends Funcionario {
    double salario;

    FuncionarioAssalariado(double salario) {
        this.salario = salario;
    }

    double calcularSalario() {
        return salario;
    }
}

public class Main {
    public static void main(String[] args) {
        Funcionario funcionario1 = new FuncionarioHorista(160, 25);
        Funcionario funcionario2 = new FuncionarioAssalariado(3000);

        System.out.println(funcionario1.calcularSalario());
        System.out.println(funcionario2.calcularSalario());
    }
}

File: Go/exercicio11.go
package main

import "fmt"

type Funcionario interface {
    CalcularSalario() float64
}

type FuncionarioHorista struct {
    HorasTrabalhadas int
    ValorPorHora     float64
}

func (f FuncionarioHorista) CalcularSalario() float64 {
    return float64(f.HorasTrabalhadas) * f.ValorPorHora
}

type FuncionarioAssalariado struct {
    Salario float64
}

func (f FuncionarioAssalariado) CalcularSalario() float64 {
    return f.Salario
}

func main() {
    funcionario1 := FuncionarioHorista{160, 25}
    funcionario2 := FuncionarioAssalariado{3000}

    fmt.Println(funcionario1.CalcularSalario())
    fmt.Println(funcionario2.CalcularSalario())
}
